### ISSUE TITLE
"Fatal error. Internal CLR error. (0x80131506)" when using ValueTask<T>

### ISSUE THREAD

**Issue by fdipuma** [Singulink/RuntimeNullables#9]
Hi! I've got an issue with the package along with `ValueTask<T>` return types.

If nullable checks are enabled, and you have a method with return type `ValueTask<anything>` (with anything not nullable), executing that method crashes the process with an `Fatal error. Internal CLR error. (0x80131506)` message.

How to reproduce (xunit test):

```csharp
using RuntimeNullables;
using Xunit;

namespace ValueTaskIssue;

public class TestedClass
{
    [NullChecks(true)]
    public ValueTask<string> TestedMethod1()
    {
        return new ValueTask<string>("");
    }
    
    [NullChecks(false)]
    public ValueTask<string> TestedMethod2()
    {
        return new ValueTask<string>("");
    }
}

public class UnitTest1
{
    private readonly TestedClass _testedClass = new TestedClass();

    // this fails with: Fatal error. Internal CLR error. (0x80131506)
    [Fact]
    public async Task Test1()
    {
        var value = await _testedClass.TestedMethod1();

        Assert.Empty(value);
    }
    
    // this works:
    [Fact]
    public async Task Test2()
    {
        var value = await _testedClass.TestedMethod2();

        Assert.Empty(value);
    }
}
```

I'm not super sure, but I suppose it could be related to the weaver trying to add nullable checks to the value of the ValueTask

**Comment by mikernet** [comment]
We have unit tests for value tasks and I have used them without a problem so this seems like something rather specific causing this.

What version of .NET are you running this on?

**Comment by mikernet** [comment]
Okay, I see the issue, I can repro. Here you are using `ValueTask` without `async`. Will post a fix shortly.

**Comment by mikernet** [comment]
Fixed in v1.0.6, should be up on nuget in a couple mins.

**Comment by fdipuma** [comment]
Thanks! That was super quick :)
